pkg/controllers/osc/resources/operating_system_config.go

```diff
@@ -71,48 +69,49 @@ func OperatingSystemConfigCreator(
 			if err != nil {
 				return nil, fmt.Errorf("failed to get cloud provider from machine deployment: %v", err)
 			}
+			cloudConfig := string(cloudProvider.Spec.Raw)
```

Comment:
I'm not sure if this direct mapping is supposed to work. Aren't we supposed to generate `cloud-config` based on the templates like these for different cloud providers?

https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/aws/types/cloudconfig.go#L30

Reply:
I think it won't work but that's fine. I have an idea to solve this after the PR is merged ;)